api/src/exam-environment/routes/exam-environment.ts

```diff
@@ -705,9 +729,71 @@ async function getExams(
         retakeTimeInMS: exam.config.retakeTimeInMS,
         passingPercent: exam.config.passingPercent
       },
-      canTake: isExamPrerequisitesMet
+      canTake: false
     };
-  });
+
+    const isExamPrerequisitesMet = checkPrerequisites(user, exam.prerequisites);
+
+    if (!isExamPrerequisitesMet) {
+      availableExam.canTake = false;
+      availableExams.push(availableExam);
+      continue;
+    }
+    // Latest attempt must be:
+    // a) Moderated
+    // b) Past exam config retake time
+    const lastAttempt = attempts.length
```

Comment:
Not sure if I have this right - thinking out loud...

`exams` is all the exams in the db?
`attempts` is **all** the users attempts? Which may have several attempts, one (or more?) for each exam.

Wouldn't last attempt here find the overall last attempt then? for any exam - not just the last attempt for the current `exam.id` in the loop?